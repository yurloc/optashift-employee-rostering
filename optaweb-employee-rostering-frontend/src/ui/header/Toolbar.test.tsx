/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { getRouterProps } from 'util/BookmarkableTestUtils';
import { ToolbarComponent, Props } from './Toolbar';

describe('Toolbar Component', () => {
  beforeAll(() => {
    process.env.REACT_APP_BACKEND_URL = 'backend';
  });

  it('should render correctly with no tenants', () => {
    const toolbarComponent = shallow(<ToolbarComponent {...noTenants} />);
    expect(toJson(toolbarComponent)).toMatchSnapshot();
  });

  it('should render correctly with a few tenants', () => {
    const toolbarComponent = shallow(<ToolbarComponent {...twoTenants} />);
    expect(toJson(toolbarComponent)).toMatchSnapshot();
  });

  it('should render correctly when tenant select is open', () => {
    const toolbarComponent = shallow(<ToolbarComponent {...twoTenants} />);
    (toolbarComponent.instance() as ToolbarComponent).setIsTenantSelectOpen(true);
    expect(toJson(toolbarComponent)).toMatchSnapshot();
  });

  it('should refresh tenant list on mount', async () => {
    shallow(<ToolbarComponent {...twoTenants} />);
    expect(twoTenants.refreshTenantList).toBeCalled();
  });

  it('should call change tenant on tenant change', () => {
    const toolbarComponent = shallow(<ToolbarComponent {...twoTenants} />);
    (toolbarComponent.instance() as ToolbarComponent).setCurrentTenant(2);
    expect(twoTenants.changeTenant).toBeCalled();
    expect(twoTenants.changeTenant).toBeCalledWith({ routeProps: twoTenants, tenantId: 2 });
  });

  it('should redirect to admin page when you click on the gear', () => {
    const toolbarComponent = shallow(<ToolbarComponent {...twoTenants} />);
    toolbarComponent.find('[aria-label="Settings"]').simulate('click');
    expect(twoTenants.history.push).toBeCalled();
    expect(twoTenants.history.push).toBeCalledWith('/admin');
  });
});

const noTenants: Props = {
  tenantList: [],
  currentTenantId: 0,
  refreshTenantList: jest.fn(),
  changeTenant: jest.fn(),
  ...getRouterProps('/toolbar', {}),
};

const twoTenants: Props = {
  tenantList: [
    {
      id: 1,
      version: 0,
      name: 'Tenant 1',
    },
    {
      id: 2,
      version: 0,
      name: 'Tenant 2',
    },
  ],
  currentTenantId: 2,
  refreshTenantList: jest.fn(),
  changeTenant: jest.fn(),
  ...getRouterProps('/toolbar', {}),
};
