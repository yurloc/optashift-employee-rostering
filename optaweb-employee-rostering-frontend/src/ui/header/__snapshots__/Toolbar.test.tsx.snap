// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Component should render correctly when tenant select is open 1`] = `
<Toolbar>
  <ToolbarGroup>
    <ToolbarItem>
      <Dropdown
        dropdownItems={
          Array [
            <DropdownItem
              data-tenantid={1}
            >
              Tenant 1
            </DropdownItem>,
            <DropdownItem
              data-tenantid={2}
            >
              Tenant 2
            </DropdownItem>,
          ]
        }
        isOpen={true}
        isPlain={true}
        onSelect={[Function]}
        position="right"
        toggle={
          <DropdownToggle
            onToggle={[Function]}
          >
            Tenant 2
          </DropdownToggle>
        }
      />
    </ToolbarItem>
  </ToolbarGroup>
  <ToolbarGroup>
    <ToolbarItem>
      <Button
        aria-label="Notifications"
        variant="plain"
      >
        <BellIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
          title={null}
        />
      </Button>
    </ToolbarItem>
    <ToolbarItem>
      <Button
        aria-label="Settings"
        data-cy="settings"
        onClick={[Function]}
        variant="plain"
      >
        <CogIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
          title={null}
        />
      </Button>
    </ToolbarItem>
    <ToolbarItem>
      <a
        aria-label="REST Reference"
        href="backend/swagger-ui.html"
      >
        <Button
          variant="plain"
        >
          <BookIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
            title={null}
          />
        </Button>
      </a>
    </ToolbarItem>
  </ToolbarGroup>
</Toolbar>
`;

exports[`Toolbar Component should render correctly with a few tenants 1`] = `
<Toolbar>
  <ToolbarGroup>
    <ToolbarItem>
      <Dropdown
        dropdownItems={
          Array [
            <DropdownItem
              data-tenantid={1}
            >
              Tenant 1
            </DropdownItem>,
            <DropdownItem
              data-tenantid={2}
            >
              Tenant 2
            </DropdownItem>,
          ]
        }
        isOpen={false}
        isPlain={true}
        onSelect={[Function]}
        position="right"
        toggle={
          <DropdownToggle
            onToggle={[Function]}
          >
            Tenant 2
          </DropdownToggle>
        }
      />
    </ToolbarItem>
  </ToolbarGroup>
  <ToolbarGroup>
    <ToolbarItem>
      <Button
        aria-label="Notifications"
        variant="plain"
      >
        <BellIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
          title={null}
        />
      </Button>
    </ToolbarItem>
    <ToolbarItem>
      <Button
        aria-label="Settings"
        data-cy="settings"
        onClick={[Function]}
        variant="plain"
      >
        <CogIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
          title={null}
        />
      </Button>
    </ToolbarItem>
    <ToolbarItem>
      <a
        aria-label="REST Reference"
        href="backend/swagger-ui.html"
      >
        <Button
          variant="plain"
        >
          <BookIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
            title={null}
          />
        </Button>
      </a>
    </ToolbarItem>
  </ToolbarGroup>
</Toolbar>
`;

exports[`Toolbar Component should render correctly with no tenants 1`] = `
<Toolbar>
  <ToolbarGroup />
  <ToolbarGroup>
    <ToolbarItem>
      <Button
        aria-label="Notifications"
        variant="plain"
      >
        <BellIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
          title={null}
        />
      </Button>
    </ToolbarItem>
    <ToolbarItem>
      <Button
        aria-label="Settings"
        data-cy="settings"
        onClick={[Function]}
        variant="plain"
      >
        <CogIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
          title={null}
        />
      </Button>
    </ToolbarItem>
    <ToolbarItem>
      <a
        aria-label="REST Reference"
        href="backend/swagger-ui.html"
      >
        <Button
          variant="plain"
        >
          <BookIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
            title={null}
          />
        </Button>
      </a>
    </ToolbarItem>
  </ToolbarGroup>
</Toolbar>
`;
